File: android/src/main/java/com/shakebugs/react/utils/Emitter.java
package com.shakebugs.react.utils;

import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.modules.core.DeviceEventManagerModule;

public class Emitter {
    public static final String EVENT_NOTIFICATION = "EventNotification";
    public static final String EVENT_UNREAD_MESSAGES = "UnreadMessages";
    public static final String EVENT_HOME_ACTION_TAP = "HomeActionTap";
    public static final String EVENT_SHAKE_OPEN = "OnShakeOpen";
    public static final String EVENT_SHAKE_DISMISS = "OnShakeDismiss";
    public static final String EVENT_SHAKE_SUBMIT = "OnShakeSubmit";
    private final ReactContext reactContext;

    public Emitter(ReactContext reactContext) {
        this.reactContext = reactContext;
    }

    public void sendEvent(String eventName, WritableMap params) {
        reactContext
                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                .emit(eventName, params);
    }

    public void sendEvent(String eventName, int number) {
        reactContext
                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                .emit(eventName, number);
    }

    public void sendEvent(String eventName, String text) {
        reactContext
                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                .emit(eventName, text);
    }
}

File: android/src/main/java/com/shakebugs/react/utils/Emitter.kt
package com.shakebugs.react.utils

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.WritableMap
import com.facebook.react.modules.core.DeviceEventManagerModule

class Emitter(private val reactContext: ReactApplicationContext) {

    fun sendEvent(eventName: String, params: WritableMap) {
        reactContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
            .emit(eventName, params)
    }

    fun sendEvent(eventName: String, number: Int) {
        reactContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
            .emit(eventName, number)
    }

    fun sendEvent(eventName: String, text: String) {
        reactContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
            .emit(eventName, text)
    }

    companion object {
        const val EVENT_NOTIFICATION: String = "EventNotification"
        const val EVENT_UNREAD_MESSAGES: String = "UnreadMessages"
        const val EVENT_HOME_ACTION_TAP: String = "HomeActionTap"
        const val EVENT_SHAKE_OPEN: String = "OnShakeOpen"
        const val EVENT_SHAKE_DISMISS: String = "OnShakeDismiss"
        const val EVENT_SHAKE_SUBMIT: String = "OnShakeSubmit"
    }
}
